File: javaproject/app/src/main/java/com/marcosvbras/robomarket/flows/createsale/ui/adapter/ItemSaleRobot.java
package com.marcosvbras.robomarket.flows.createsale.ui.adapter;

import android.view.View;

import com.genius.groupie.Item;
import com.marcosvbras.robomarket.R;
import com.marcosvbras.robomarket.business.domain.ItemRobotQuantity;
import com.marcosvbras.robomarket.databinding.ItemRobotSaleBinding;
import com.marcosvbras.robomarket.interfaces.OnRecyclerClick;

public class ItemSaleRobot extends Item<ItemRobotSaleBinding> implements View.OnClickListener {

    private ItemRobotQuantity itemRobotQuantity;
    private OnRecyclerClick onRecyclerClick;

    public ItemSaleRobot(ItemRobotQuantity itemRobotQuantity, OnRecyclerClick onRecyclerClick) {
        this.itemRobotQuantity = itemRobotQuantity;
        this.onRecyclerClick = onRecyclerClick;
    }

    @Override
    public int getLayout() {
        return R.layout.item_robot_sale;
    }

    @Override
    public void bind(ItemRobotSaleBinding viewBinding, int position) {
        viewBinding.setItemRobotQuantity(itemRobotQuantity);
        viewBinding.getRoot().setOnClickListener(this);
    }

    @Override
    public void onClick(View v) {
        onRecyclerClick.onClick(itemRobotQuantity);
    }
}

File: javaproject/app/src/main/java/com/marcosvbras/robomarket/flows/createsale/ui/adapter/ItemSaleRobot.kt
package com.marcosvbras.robomarket.flows.createsale.ui.adapter

import android.view.View
import com.genius.groupie.Item
import com.marcosvbras.robomarket.R
import com.marcosvbras.robomarket.business.beans.ItemRobotQuantity
import com.marcosvbras.robomarket.databinding.ItemRobotSaleBinding
import com.marcosvbras.robomarket.interfaces.OnRecyclerClick

class ItemSaleRobot(private val itemRobotQuantity: ItemRobotQuantity, private val onRecyclerClick: OnRecyclerClick) : Item<ItemRobotSaleBinding>(), View.OnClickListener {

    override fun getLayout(): Int {
        return R.layout.item_robot_sale
    }

    override fun bind(viewBinding: ItemRobotSaleBinding, position: Int) {
        viewBinding.itemRobotQuantity = itemRobotQuantity
        viewBinding.root.setOnClickListener(this)
    }

    override fun onClick(v: View) {
        onRecyclerClick.onClick(itemRobotQuantity)
    }

}